src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import (division, print_function)
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+from pprint import pformat
+
+
+class DDMReqTemplate(dict):
+    """
+    A simple class defining a set of request templates for DDM, with default
+    structures representing the different POST calls expected by the modules
+    here:
+        https://github.com/SmartDataProjects/dynamo/tree/master/lib/web/modules
+    """
+
+    def __init__(self, api=None, **kwargs):
+        """
+        api:
+            String representing the type of request to be sent. In case the api
+            is missing or unrecognized the created object will be an ordinary
+            dictionary.
+        example:
+            ddmReq = DDMReqTemplate('copy',
+                                     item=['/LQLQToTop..._M-1200/NANOAODSIM'],
+                                     site=['T2_CH_CERN', 'T2_US_MIT'])
+        NOTE: we do not enforce the use of api parameter, but we may change this
+        """
+        super(DDMReqTemplate, self).__init__(**kwargs)
+
+        self.api = api
+        if api is None:
+            msg = "ERROR: Missing API parameter"
+            raise ValueError(msg)
+
+        # tempTuple format: [(key, defaultValue, type), ....()]
+        if api == 'copy':
+            tempTuple = [('item', [], list),
+                         ('site', ['T2_*', 'T1_*_Disk'], list),
+                         ('group', 'DataOps', str),
+                         ('n', None, int),
```

Comment:
The default values are not type-checked so the `None` values here do not break the checks bellow. This is done in order to always have a static structure of the template with the unneeded fields set to `None` until the very last moment when they are dropped just before sending them to DDM.
   
`n` - should be the number of copies we request. About the `cache` - we  actually did not get a  

`cache` - I'd better just cite the answer from DDM developers:

```
Finally, cache=y as parameter puts it into another table of dynamo to cache them, which dynamo looks at every two hours. This was done because somehow without the cache=y, at some point the call to the API took very long for unified and this was a way around it. I would like to try it without the cache=y first, but it's definitely an option too.
```